dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientChannels.java

```diff
@@ -394,7 +395,12 @@ public CloseableClient build() {
 
             HttpClientBuilder builder = HttpClients.custom()
                     .setDefaultRequestConfig(RequestConfig.custom()
-                            .setConnectTimeout(connectTimeout)
+                            // HTTPCLIENT-1478: Work around the connection timeout being used to negotiate proxy
+                            // traffic.
+                            .setConnectTimeout(
+                                    ProxySelector.getDefault().equals(conf.proxy())
+                                            ? connectTimeout
+                                            : Timeout.ofMilliseconds(socketTimeoutMillis))
```

Comment:
this block is the non-test change.

Reply:
Ok so just to make sure i understand - this behaviour was working fine on hc4 and seems to be borked on hc5, which is why we've got this workaround in one place only?

Reply:
That’s correct, I opened an upstream PR to fix the bug a few minutes ago as well.